packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts

```diff
@@ -488,66 +489,96 @@ class BaseDescopeWc extends HTMLElement {
     };
   }
 
-  async #loadDescopeUI() {
-    const SCRIPT_ID = 'load-descope-ui';
+  async #getComponentsVersion() {
+    const version = (await this.#getConfig())?.projectConfig?.componentsVersion;
 
-    // if DescopeUI is already loaded, use it
-    if (globalThis.DescopeUI) {
-      this.descopeUI = Promise.resolve(globalThis.DescopeUI);
-      return;
-    }
+    if (version) return version;
 
-    const existingScript = document.getElementById(
-      SCRIPT_ID,
-    ) as HTMLScriptElement;
-    const scriptEle: HTMLScriptElement =
-      existingScript || document.createElement('script');
+    this.logger.error('Did not get components version, using latest version');
 
-    if (existingScript) {
+    return 'latest';
+  }
+
+  static descopeUI: any;
+
+  async #loadDescopeUI() {
+    if (BaseDescopeWc.descopeUI) {
       this.loggerWrapper.info(
-        'DescopeUI loading script is already exist, probably multiple flows are running on the same page',
+        'DescopeUI is already loading, probably multiple flows are running on the same page',
       );
-    } else {
-      scriptEle.id = SCRIPT_ID;
-
-      let version = (await this.#getConfig())?.projectConfig?.componentsVersion;
+      return;
+    }
 
-      if (!version) {
-        this.logger.error(
-          'Did not get components version, using latest version',
-        );
-        version = 'latest';
+    BaseDescopeWc.descopeUI = new Promise((resolve) => {
+      if (globalThis.DescopeUI) {
+        resolve(globalThis.DescopeUI);
+        return;
       }
 
-      scriptEle.src = UI_COMPONENTS_URL.replace(
-        UI_COMPONENTS_URL_VERSION_PLACEHOLDER,
-        version,
-      );
+      const setupScript = (url: string) => {
+        const scriptEle = document.createElement('script');
+        scriptEle.id = 'load-descope-ui';
+        scriptEle.src = url;
 
-      document.body.append(scriptEle);
-    }
+        return scriptEle;
+      };
 
-    this.descopeUI = new Promise((res) => {
-      const onError = () => {
-        this.loggerWrapper.error(
-          'Cannot load DescopeUI',
-          `Make sure this URL is valid and return the correct script: "${scriptEle.src}"`,
+      const generateScriptUrl = (
```

Comment:
great abstraction and separation of concerns to functions! not critical but maybe we should put them in a util file cuz they don't use `this` or anything from the current scope

Reply:
Thanks!

I thought about it as well, but it's very specific to this implementation and I don't think we will use it elsewhere

That's why I decided to leave it here

If you think we should move it into a separate file, 
I can do it in 2 min

LMK what you think